Pagination/Configuration/PaginateFinderConfiguration.php

```diff
@@ -14,26 +14,41 @@ class PaginateFinderConfiguration extends FinderConfiguration
     protected $limit = null;
 
     /**
+     * @param null|array          $order
+     * @param null|int            $skip
+     * @param null|int            $limit
+     */
+    public function setPaginateConfiguration($order = null, $skip = null, $limit = null)
+    {
+        $this->setLimit($limit);
+        if($this->isArrayOrNull($order)){
+            $this->setOrder($order);
+        }
+        $this->setSkip($skip);
+    }
+
+    /**
      * @param Request $request
      *
      * @return PaginateFinderConfiguration
      */
     public static function generateFromRequest(Request $request)
     {
-        $finderConfig = new PaginateFinderConfiguration();
+        $configuration = new static();
+
         $columns = $request->get('columns');
-        if(PaginateFinderConfiguration::isArrayOrNull($columns)){
-            $finderConfig->setColumns($columns);
-        }
-        $finderConfig->setSearch($request->get('search'));
-        $finderConfig->setLimit($request->get('limit'));
-        $order = $request->get('order');
-        if(PaginateFinderConfiguration::isArrayOrNull($order)){
-            $finderConfig->setOrder($order);
+        if ($configuration->isArrayOrNull($columns)) {
```

Comment:
yes, but with new self() or new static(), it will return object of various kind.